UI/Extensions/SearchModelExtensions.cs

```diff
@@ -38,12 +27,56 @@ public static string ToQueryString(this SearchModel? model)
             return dictionary;
         }
 
+        private static Dictionary<string, string> ToRouteData(this SearchModel model, bool flattenCollection)
+        {
+            var dictionary = new Dictionary<string, string>();
+
+            model.Add(x => x.From, dictionary);
+            model.Add(x => x.Name, dictionary);
+            model.Add(x => x.Postcode, dictionary);
+            model.Add(x => x.TuitionType, dictionary);
+            model.AddAll(x => x.KeyStages, dictionary, flattenCollection);
+            model.AddAll(x => x.Subjects, dictionary, flattenCollection);
+            model.Add(x => x.ShortlistOrderBy, dictionary);
+            model.Add(x => x.ShortlistOrderByDirection, dictionary);
+            model.Add(x => x.ShortlistTuitionType, dictionary);
+            model.Add(x => x.ShortlistGroupSize, dictionary);
+
+            return dictionary;
+        }
+
         private static void AddAll<T>(
             this SearchModel model,
             Expression<Func<SearchModel, IEnumerable<T>?>> expression,
-            Dictionary<string, string> dictionary)
+            Dictionary<string, string> dictionary,
+            bool flattenCollection)
         {
-            model.Add(expression, (prop, name) => BuildQueryString(prop, name), dictionary);
+            if (flattenCollection)
+            {
+                model.Add(expression, (prop, name) => BuildQueryString(prop, name), dictionary);
```

Comment:
`model.Add(expression, (prop, name) => BuildQueryString(prop, name), dictionary);` can be converted into a method group by replacing the lambda expression with a method group, like so:
 `model.Add(expression, BuildQueryString, dictionary);` It is less code.

Reply:
Done